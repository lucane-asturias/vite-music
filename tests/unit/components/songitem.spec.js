import SongItem from '@/components/SongItem.vue'
import { shallowMount, RouterLinkStub } from '@vue/test-utils'

describe('Tests on the SongItem.vue component', () => {
  const song = { 
    display_name: 'test',
    docID: 'INDI9IiÃ§Jm!xRNIOQRO'
  }

  const wrapper = shallowMount(SongItem, {
    props: { song },
    global: {
      components: { 'router-link': RouterLinkStub },
      mocks: { $t: (translation) => translation },
    },
  })

  test('must match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders song.display_name', () => {
    const compositionAuthor = wrapper.find('.text-gray-500')

    expect(compositionAuthor.text()).toBe(`song.artist ${song.display_name}`)
  })

  test('renders song.docID in id attribute', () => {
    // expect(wrapper.findComponent(RouterLinkStub).componentVM.to.params.id).toBe(song.docID)

    expect(wrapper.findComponent(RouterLinkStub).props().to).toEqual({ 
      name: 'song', 
      params: { id: song.docID }, 
    })
  })

})