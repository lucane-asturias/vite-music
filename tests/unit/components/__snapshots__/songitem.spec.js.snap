// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongItem.vue > must match snapshot 1`] = `
"<li class=\\"flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50\\" data-v-c890381d=\\"\\">
  <div data-v-c890381d=\\"\\">
    <!-- /song/jW3N2QiCpFi4sply21oC -->
    <router-link-stub to=\\"[object Object]\\" custom=\\"false\\" class=\\"font-bold block text-gray-600\\" data-v-c890381d=\\"\\"></router-link-stub><span class=\\"text-gray-500 text-sm\\" data-v-c890381d=\\"\\">song.artist test</span>
  </div>
  <div class=\\"text-gray-600 text-lg\\" data-v-c890381d=\\"\\">
    <router-link-stub to=\\"[object Object]\\" custom=\\"true\\" data-v-c890381d=\\"\\"></router-link-stub>
  </div>
</li>
<div class=\\"space\\" data-v-c890381d=\\"\\"></div>"
`;

exports[`Tests on the SongItem.vue component > must match snapshot 1`] = `
"<li class=\\"flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50\\" data-v-c890381d=\\"\\">
  <div data-v-c890381d=\\"\\">
    <!-- /song/jW3N2QiCpFi4sply21oC -->
    <router-link-stub to=\\"[object Object]\\" custom=\\"false\\" class=\\"font-bold block text-gray-600\\" data-v-c890381d=\\"\\"></router-link-stub><span class=\\"text-gray-500 text-sm\\" data-v-c890381d=\\"\\">song.artist test</span>
  </div>
  <div class=\\"text-gray-600 text-lg\\" data-v-c890381d=\\"\\">
    <router-link-stub to=\\"[object Object]\\" custom=\\"true\\" data-v-c890381d=\\"\\"></router-link-stub>
  </div>
</li>
<div class=\\"space\\" data-v-c890381d=\\"\\"></div>"
`;
